### PR TITLE
fix(developer): prevent invalid values in targets store

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes: #11773

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by DavidLRowe** [comment]
Not super important, but if you're making changes in this area anyway...
Would you consider adding a warning for cases that are not wrong, but perhaps confusing? I'm thinking primarily of:
- any windows macos linux web ... (where the user has checked all the boxes)

but maybe "any" plus another box would also fall in that category.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by mcdurdin** [comment]
@DavidLRowe I've created #11996 for this but pushed it to a future release because I have no capacity to do it at present.

File: developer/src/common/include/kmn_compiler_errors.h [hunk 1]
```diff
@@ -181,6 +181,9 @@
 #define CERR_CharacterRangeTooLong                         0x00004078
 #define CERR_NonBMPCharactersNotSupportedInKeySection      0x00004079
 
+#define CERR_InvalidTarget                                 0x00004079
+#define CERR_NoTargetsSpecified                            0x0000407A
```
Comment (mcdurdin) [inline developer/src/common/include/kmn_compiler_errors.h:185]:
```suggestion
#define CERR_InvalidTarget                                 0x0000407A
#define CERR_NoTargetsSpecified                            0x0000407B
```

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.73-alpha](https://keyman.com/downloads/releases/alpha/18.0.73)

**Pull request merged**